omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,213 @@
+			}
+		}
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("name");
+		description.addProperty("description");
+		description.addProperty("identifierType");
+		
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("name");
+		description.addProperty("description");
+		description.addProperty("identifierType");
+		
+		return description;
+	}
+	
+	@PropertyGetter("identifierType")
+	public String getidentifierType(IdentifierSource identifierSource) {
+		return identifierSource.getIdentifierType().getName();
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(IdentifierSource identifierSource) {
+		return identifierSource.getIdentifierType() + " - " + identifierSource.getName();
+	}
+	
+	@Override
+	public IdentifierSource newDelegate() {
+		return new SequentialIdentifierGenerator();
+	}
+	
+	@Override
+    public Object create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		
+		Object savedIdentifierSource = null;
+		
+		if(postBody.get("sourceType").toString().equals("SequentialIdentifierGenerator")) {
+			
+			SequentialIdentifierGenerator identifierSource = new SequentialIdentifierGenerator();
+			
+			String uuid = postBody.get("identifierType").toString();
+			String name = postBody.get("name").toString();
+			String description = postBody.get("description").toString();
+			String firstIdentifierBase = postBody.get("firstIdentifierBase").toString();
+			String baseCharacterSet = postBody.get("baseCharacterSet").toString();
+			String prefix = postBody.get("prefix").toString();
+			String suffix = postBody.get("suffix").toString();
+			Integer minLength = Integer.parseInt(postBody.get("minLength").toString());
+			Integer maxLength = Integer.parseInt(postBody.get("maxLength").toString());
+			
+			identifierSource.setIdentifierType(Context.getPatientService().getPatientIdentifierTypeByUuid(uuid));
+			identifierSource.setName(name);
+			identifierSource.setDescription(description);
+			identifierSource.setBaseCharacterSet(baseCharacterSet);
+			identifierSource.setFirstIdentifierBase(firstIdentifierBase);
+			identifierSource.setPrefix(prefix);
+			identifierSource.setSuffix(suffix);
+			identifierSource.setMinLength(minLength);
+			identifierSource.setMaxLength(maxLength);
+			
+			savedIdentifierSource =  save(identifierSource);	
+		}
+		
+		else if(postBody.get("sourceType").toString().equals("RemoteIdentifierSource")) {
+			
+			RemoteIdentifierSource identifierSource = new RemoteIdentifierSource();
+			
+			String uuid = postBody.get("identifierType").toString();
```

Comment:
The uuid variable name does not look as clear as something like identifierTypeUuid

Reply:
fixed